### PR TITLE
Refs #37443 - find the right Edit button on the host details page

### PR DESCRIPTION
**Pull request by evgeni**
Fixes: 37093a54fd36ab247c809d86420ea341f4630ac7

### CODE REVIEW

File: test/integration/host_js_test.rb [hunk 1]
```diff
@@ -529,6 +529,7 @@ class HostJSTest < IntegrationTestWithJavascript
       page.find(id).click
       assert page.has_no_selector?(id)
       click_button('Submit')
+      find('h5', :text => host.fqdn) # wait for the host page to load
```
Comment (evgeni) [inline test/integration/host_js_test.rb]: There is a dedicated `click_on_submit` helper (introduced in 1580adbec7995b3d27e55d4067a2d5a4fd34d291), but all of its uses were removed in 98c17528a655148bf4c01629ea36cf249441c213 and I am not sure why.

Following the same logic as in 98c17528a655148bf4c01629ea36cf249441c213 and using a `find` to find the header with the hostname.

Comment (evgeni) [reply test/integration/host_js_test.rb]: Split that change out to https://github.com/theforeman/foreman/pull/10302

**Comment by evgeni** [comment]
#10298 and #10302 were tested together in https://github.com/theforeman/foreman/pull/10125 and turn up green \o/

**Review by MariaAga — APPROVED** [review]
LGTM, thank you!

**Pull request merged**